### PR TITLE
Brain damage crit

### PR DESCRIPTION
**Pull request by Superlagg**
[FEATURE][BALANCE][INPUT WANTED][DRAFT]

## About the PR 

Changes the way brain damage works from a lengthy stun at ~100 and instant death at 120, to instant death at 220 and horrible debilitating conditions starting at 50 brain damage, as well as a chance for headaches at around 10 brain damage.

Brain damage is weighted considerably less for the death chance, since now it can go well above 120.

Synaptizine is adjusted to only work below 180 brain damage.

Added three new maladies:
 - Headache - Starts showing up at 10 brain damage, throws a few messages about how painful it is, and inflicts various annoying yet harmless conditions, such as blurred vision and stuttering. Mainly here to make it clear to the player that they've taken brain damage. Treated with pain meds and synaptizine.
- Traumatic Brain Injury - Starts showing up at 50 brain damage, throws a bunch of scary messages about your head, inflicts several harmful conditions, such as random stuns, slowed movement, bleeding, and further brain damage. Intended to be basically heart failure, but brain. Treated with various stimulants, mannitol, and synaptizine. Can progress to...
- Coma - Starts showing up at 100 brain damage. Locked in a weakened state with accumulating brain and oxy damage. Brain damage continues until around 180, at which point it'll only have a 10% chance to cause damage. Treated with synaptizine, mannitol, and coffee (sort of). Can progress to heart failure.

Player description is altered to account for the new brain damage levels.

Medscanners are also more accurate about how damaged the patient's brain is.

## Why's this needed? 

Brain damage is weird. On one hand, there's little indication that you've been brainhurt (at least compared to crit), and on the other it'll instantly kill you if it gets too high. 

That's no fun! So instead of just straight up dying once it starts getting interesting, brain damage gets its own form of crit, which will totally kill you if it gets too bad, and makes sure you absolutely know you're dying in the process.

Death should really be the grand finale to a dramatic build up, which crit does pretty well! Brain damage just kinda tacks it on wherever it feels like, and too often right before anything interesting happens. 

## Changelog

```
(u)Superlagg:
(*)Brain damage now kills at 220 brain damage instead of 120
(*)Your brain can now go into crit! It starts at around 50 damage, with a chance to slip into a coma past 100 damage.
```

### CODE REVIEW

**Comment by zjdtmkhzt** [comment]
The meathead trait will probably need some adjustments, since I think it gives a permanent 60 brain damage right now?
Having constant traumatic brain injury symptoms sounds like it would really devalue that trait a LOT, maybe possibly even enough to make it a 0 point trait?

**Comment by Tarmunora** [comment]
My thoughts: brain damage mixes are lethal enough without having brain damage cause itself past 50, and extending the brain damage permastun from 100-120 brain to 100-220 brain would just serve to draw out deaths, past the point where the victim has any agency left (they get to stare at their KOed body for 3 minutes instead of 2 while they succumb or wait to die)

File: code/mob/living/carbon.dm [hunk 1]
```diff
@@ -176,14 +176,45 @@
...
 		src.visible_message("<span class='alert'><b>[src.name]</b> goes limp, their facial expression utterly blank.</span>")
 		src.death()
 		return
+	
+	if (src.get_brain_damage() >= 100) // Coma
+		if (!is_chg)
+			boutput(src, "<span class='alert'>You feel [pick("oddly", "suddenly", "quite")] [pick("faint", "dizzy", "numb", "dazed")].</span>")
+			src.change_misstep_chance(10)
+			if (!src.find_ailment_by_type(/datum/ailment/malady/coma))
+				if (prob(src.get_brain_damage() * -0.5))
```
Comment (zjdtmkhzt) [inline code/mob/living/carbon.dm:193]: I don't understand why you invert the brain damage in your probs, it's not normally negative, is it?

Comment (Superlagg) [reply code/mob/living/carbon.dm]: My bad, I copied from the deathchance and forgot to remove the negative. Fixed, though!

File: code/mob/living/carbon.dm [hunk 2]
```diff
@@ -176,14 +176,45 @@
...
+					src.emote(pick("faint", "collapse"))
+					src.visible_message("<span class='alert'><b>[src.name]</b> suddenly falls limp!</span>")
+					boutput(src, "<span class='alert'><b>You sieze up and fall limp, the pain in your head fading along with your consciousness.</b></span>")
```
Comment (zjdtmkhzt) [inline code/mob/living/carbon.dm:196]:
```suggestion
					boutput(src, "<span class='alert'><b>You seize up and fall limp, the pain in your head fading along with your consciousness.</b></span>")
```

File: code/modules/medical/diseases/braincrit.dm [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+				boutput(affected_mob, "<span class='alert'>You sneeze, spraying blood everywhere!</span>")
+				affected_mob.bleed(affected_mob, 2, 2)
+			if (prob(8))
+				affected_mob.emote("gasp")
+				affected_mob.losebreath+=3
+			if (prob(12))
+				boutput(affected_mob, "<span class='alert'>You feel [pick("numb", "confused", "dizzy", "dazed")].</span>")
+				affected_mob.drowsyness += 2
+			if (prob(1) && prob(1) && prob(1))	// Thanks, find+replace!
+				boutput(affected_mob, "<span class='alert'>Your brain stops beating!</span>")
```
Comment (zjdtmkhzt) [inline code/modules/medical/diseases/braincrit.dm:76]: I can't tell if this is intentional, but it's funny.^^

**Comment by TDHooligan** [comment]
Yeah, I agree with tarm's points, my other two cents:

I totally think that brain damage should be more noticeable early on and at definitely debilitating at the 60+ mark, but its ability to be used as a heavy weight on death rolls in itself is a good consideration for poisons, even those not going solely for brain damage deaths.

As a matter of fact, brain poisons will probably be nerfed significantly with just the warning signs, as even a prime brain poison is quite slow owing to the small number of brain poisons in the game right now.

File: code/mob/living/carbon.dm (hunk 2 above)
Comment (zjdtmkhzt) [reply code/mob/living/carbon.dm]: Oh god, why is it attaching these reviews to the first one I made, I have never done this before, why are you doing this to me, github.

I think this is spelled "seize"?
And brain damage should not be inverted, I think!

**Comment by Superlagg** [comment]
> My thoughts: brain damage mixes are lethal enough without having brain damage cause itself past 50, 

Good point, TBI doesn't, itself, cause brain damage anymore.

>and extending the brain damage permastun from 100-120 brain to 100-220 brain would just serve to draw out deaths, past the point where the victim has any agency left (they get to stare at their KOed body for 3 minutes instead of 2 while they succumb or wait to die)

I disagree, for the most part. While it will do just that if they're left to die in a hidden spot, if they're found they'll have much more of a chance to be resuscitated. This way, extending it much further, there's a lot more room for something to be done about it, as opposed to outright dying if a few rolls go poorly.

As for agency, I feel like my agency would be robbed even more so through a sudden death than I would sitting in darkness for several minutes. There, I feel like someone could possibly find my spaceman before its too late, and if that seems like it wont happen, I have the choice of just succumbing and complaining about the crew.

> I totally think that brain damage should be more noticeable early on and at definitely debilitating at the 60+ mark, but its ability to be used as a heavy weight on death rolls in itself is a good consideration for poisons, even those not going solely for brain damage deaths.
>
> Additionally, brain poisons will probably be nerfed significantly with just the warning signs, as even a prime brain poison is quite slow owing to the small number of brain poisons in the game right now.

Brain damage still weighs into death rolls, just that it's supposed to be re-weighted to accommodate the deeper braincrit. I also think that stealth poisons like that shoudn't be stealthy up until they're dead, instead just to be stealthy until they're too jacked up to deal with it. After all, it just seems silly to not know that you're dying until you're dead... unless that was the whole point of the poison.

File: code/modules/medical/diseases/braincrit.dm (hunk 3 above)
Comment (Superlagg) [reply code/modules/medical/diseases/braincrit.dm:76]: =3

**Comment by TDHooligan** [comment]
> I disagree, for the most part. While it will do just that if they're left to die in a hidden spot, if they're found they'll have much more of a chance to be resuscitated. This way, extending it much further, there's a lot more room for something to be done about it, as opposed to outright dying if a few rolls go poorly.

The problem is this is practically the same as old crit, which was changed for a reason. Players will have the choice to sit staring at their screen hoping they could get revived for several minutes.

> Brain damage still weighs into death rolls, just that it's supposed to be re-weighted to accommodate the deeper braincrit. I also think that stealth poisons like that shoudn't be stealthy up until they're dead, instead just to be stealthy until they're too jacked up to deal with it. After all, it just seems silly to not know that you're dying until you're dead... unless that was the whole point of the poison.

I don't mean to say that it was worse because of the change - just that with this, BRAIN will actually be a fairly insignificant damage type unless you stack your poison towards it

**Pull request closed without merging**